benchmark-mbedTLS-vector.sh

```diff
@@ -171,35 +170,56 @@ printf "cmac algorithm\n"
 ./test_suite_cmac
 
+cd ../..
+tBuild8=$(date +%s%3N)
+
 #===============================================================================
 
-printf "\nStep 6 done.\n"
+printf "\nStep 4 done.\n"
 printf "=============================================================\n\n"
 printf "##################################################\n"
-printf "7/7. Verifying the speedup of the vectorized version ...\n"
+printf "5/5. Verifying the speedup ...\n"
 printf "##################################################\n"
 read -p "Press enter to continue"
 printf "\n"
-cd ../..
 
+tDeploy0=$(date +%s%3N)
 printf "\n++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n"
 printf "aes_xts algorithm\n"
 printf "++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n"
 build/programs/test/benchmark aes_xts
-printf "original done\n"
-printf "************************************************************\n"
-buildVec/programs/test/benchmark aes_xts
-printf "vectorized done\n"
-
 printf "\n++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n"
 printf "cmac algorithm\n"
 printf "++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n"
 build/programs/test/benchmark aes_cmac
 printf "original done\n"
-printf "************************************************************\n"
+
+tDeploy1=$(date +%s%3N)
+printf "\n++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n"
+printf "aes_xts algorithm\n"
+printf "++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n"
+buildVec/programs/test/benchmark aes_xts
+printf "\n++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n"
+printf "cmac algorithm\n"
+printf "++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n"
 buildVec/programs/test/benchmark aes_cmac
 printf "vectorized done\n"
-
+tDeploy2=$(date +%s%3N)
 #===============================================================================
 
-printf "\nDone.\n"
+tSource=$(bc -l <<< "($tSource1 - $tSource0) /1000")
+tBuildO=$(bc -l <<< "($tBuild1 - $tBuild0) /1000")
+tCodee=$(bc -l <<< "($tBuild3 - $tBuild2) /1000")
+tBuildC=$(bc -l <<< "($tBuild5 - $tBuild4) /1000")
+tTestO=$(bc -l <<< "($tBuild7 - $tBuild6) /1000")
+tTestC=$(bc -l <<< "($tBuild8 - $tBuild7) /1000")
+tDeployO=$(bc -l <<< "($tDeploy1 - $tDeploy0) /1000")
+tDeployC=$(bc -l <<< "($tDeploy2 - $tDeploy1) /1000")
+
+printf "\nStep               \tWithout Codee\tWith Codee\n"
+printf "=====================\t=============\t==========\n"
+printf "CI: SS: Clone        \t%.3f s\t\t%.3f s\n" $tSource $tSource
+printf "CI: BS: pwdirectives \t--- s\t\t%.3f s\n" $tCodee
+printf "CI: BS: make all     \t%.3f s\t\t%.3f s\n" $tBuildO $tBuildC
+printf "CI: BS: Test aes/cmac\t%.3f s\t\t%.3f s\n" $tTestO $tTestC
+printf "CI: DS: Benchmark    \t%.3f s\t\t%.3f s\n" $tDeployO $tDeployC
```

Comment:
I think it is useless to compare the execution time here, since these MbedTLS benchmarks are not benchmarking time, but throughput instead.
These tests are calculating the amount of operations that are performed per unit of time. In this case, if I am not wrong, the benchmarks are implemented with a fixed time and a variable input workload.
Comparing raw times would be correct in exec. time benchmarks, where the time is variable, and the input workload is fixed.
We need to compare throughput metrics, not raw time numbers. For example, the average throughput, the average cycles/byte, etc.